x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1316,7 +1331,7 @@ private boolean findConjunctiveComparisonInRange(BinaryComparison main, List<Ran
                                     ranges.remove(i);
                                     ranges.add(i,
                                             new Range(other.source(), other.value(),
-                                                    main.right(), lowerEq ? true : other.includeLower(),
+                                                    main.right(), lowerEq ? false : main instanceof GreaterThanOrEqual,
```

Comment:
That's a big miss - were there no tests for this?

Reply:
It seemed much like a c&p fallout.
There wasn't a test, no, so I've simply implemented the test listed as comment in the source (`2 < a AND (2 <= a < 3) -> 2 < a < 3` / `testCombineBinaryComparisonsAndRangeLower()`)